internal/pkg/cycle/until.go

```diff
@@ -25,9 +27,10 @@ func UntilError(f func() error, interval time.Duration, attempts Limit) {
 	for {
 		err := f()
 		if err != nil {
-			if counter >= attempts {
+			if !strings.Contains(err.Error(), "connection") || counter >= attempts {
```

Comment:
It is not obvious from the package name and function name that such a condition will be required. It seems that it is very specific to a small number of cases and violates the original generality.

Reply:
UntilError is used when trying to retrieve data from the exporter or when accessing the database. Both there and there may be connection problems. In these cases, we must retry. In other cases (for example, when our table structure is incorrect or something else), there is no need to retry. Implemented this in code

Reply:
renamed UntilError -> UntilConnectionError